### ISSUE TITLE
segfault on unit tests

### ISSUE THREAD

**Issue by khufkens** [computationales/rsofun#61]
I'm tracing the segfault on unix systems as shown here:
https://github.com/computationales/rsofun/runs/5572488968?check_suite_focus=true

Fails when running checks works when running testthat in terminal on linux!?

**Comment by khufkens** [comment]
Segfault traces to memory issue in LM3PPA since this commit:

https://github.com/computationales/rsofun/pull/60/commits/84277ea80ea833390cbdab6e99cd457610835a0b

**Comment by stineb** [comment]
Resolved the bug described above by pull request #63. 
On mac, rsofun now compiles, runs, and creates identical output (comparing to an old commit of before Dec 26 where the above bug was introduced).

**Comment by stineb** [comment]
There are remaining errors when running tests on Ubuntu. Error message:
```
address 0x5607748b8044, cause 'memory not mapped'
  malloc(): corrupted top size
```
According to [this](https://kb.iu.edu/d/aqsj), it could be a problem with the array sizes passed from R to C and then to Fortran (see `wrappers.c`). Worth checking them carefully again. 

It may also be worth compiling the Fortran code with additional flags to catch problems at compilation. See [here](https://stat.ethz.ch/pipermail/r-help/2008-March/156516.html), or [here](https://scicomp.stackexchange.com/questions/36045/ifort-everithing-ok-but-with-gfortran-segmentation-fault).

**Comment by khufkens** [comment]
This has to do with this line here and the `tf_base` parameter which scales respiration as far as I can tell:

https://github.com/computationales/rsofun/blob/6f48cdc52bbc45574365c3b6c4a0ced47c8891c0/src/vegetation_lm3ppa.mod.f90#L69

When the range of values of priors is defined and the value is too small, I think you get some sort of runaway process (where respiration grows C pools. I've not managed to trace the whole stack.

In short, when picking priors the lower limit should not be 0 as the priors set by BT seems to overshoot this boundary at times given the above described issue. So BT lower limits for optimization are now set at 0.1 not 0 - this works locally, where it would fail when set to 0 (as before).

**Comment by khufkens** [comment]
Outstanding issues but not critical:

- why does a value < 0 escalate into a memory issue (potential C pool overrun or something?)
- best to hard limit this parameter to a range of 0 - 1 in the Fortran code if so fickle
